src/layers/arc-layer/arc-layer.js

```diff
@@ -75,10 +86,25 @@ export default class ArcLayer extends Layer {
         drawMode: 'LINE_STRIP',
         vertices: new Float32Array(vertices)
       }),
-      instanced: true
+      instanced: true,
+      onBeforeRender: this.setLineWidth(gl, this.props.strokeWidth),
+      onAfterRender: this.setOldLineWidth(gl)
```

Comment:
Returning generated functions is always a little tricky, apart from being a little harder to reason about, there is also the danger of binding/holding on to a stale strokeWidth as the layers are recycled. You can probably avoid it by just copying strokeWidth to model.userData. And you could try to write the methods here in model directly. Also, the model should have gl context on the program member, no need to pass it in.

instanced: true,
onBeforeRender() {
  const {gl} = this.program;
  this.userData.oldStrokeWidth = gl.getParameter(...);
  gl.setParameter(this.userData.strokeWidth);
}
onAfterRender() {
  const {gl} = this.program;
  gl.setParameter(this.userData.oldStrokeWidth);
}
```
And in `willReceiveProps`:
```
this.model.userData.strokeWidth = this.props.strokeWidth;
``
No need for if statements.

You would also set model.userData.strokeWidth on initialization.